modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java

```diff
@@ -255,13 +261,15 @@ public void gatherLocalObjectStatisticsAsync(
         StatisticsGatheringContext gCtx = currColls.computeIfAbsent(gatId, k ->
             new StatisticsGatheringContext(gatId, keysSet, partsCnt));
 
-        statGathering.collectLocalObjectsStatisticsAsync(reqId, keysSet, parts, () -> gCtx.doneFut().isCancelled());
+        statGathering.collectLocalObjectsStatisticsAsync(reqId, keysSet, parts, () -> gCtx.doneFuture().isCancelled());
     }
 
     /** {@inheritDoc} */
-    @Override public StatisticsGatheringFuture<Map<StatisticsTarget, ObjectStatistics>>[] collectObjectStatisticsAsync(
+    @Override public StatisticsGatheringFuture<Map<StatisticsTarget, ObjectStatistics>>[] gatherObjectStatisticsAsync(
         StatisticsTarget... keys
     ) throws IgniteCheckedException {
+        checkSupport("collect statistics async");
```

Comment:
it's better to complete a future with this exception

Reply:
But sometimes we'll throw exception while sometimes (as you suggest) - put the same error into the array of futures? And we still can produce IgniteChechedException while starting the task if there are no such object in scheme (now).
From this point of view - what if we'll throw IgniteChechedException here if there are some node(s) without such future, but return errors about object lacks in related to such object future?

Reply:
fixed